### PR TITLE
Instagram Video and IGTV downloader

### PR DESCRIPTION
**Pull request by Epic-R-R**
Download Video and IGTV from Instagram

### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: web_programming/instagram_video.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import json
```
Comment (cclauss) [inline web_programming/instagram_video.py:1]:
```suggestion
```

File: web_programming/instagram_video.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+from datetime import datetime
+
+import requests
+
+if __name__ == "__main__":
+    link = input("Enter Video/IGTV url: ").strip()
+    url = f"https://downloadgram.net/wp-json/wppress/video-downloader/video?url={link}"
+    req = requests.get(url)
+    res = req.content
+    json_data = json.loads(res)
```
Comment (cclauss) [inline web_programming/instagram_video.py:11]:
```suggestion
    json_data = requests.get(url).json()
```

**Review by cclauss — CHANGES_REQUESTED** [review]
Requests has json builtin.

The first thing that I would do after running this would be to rename the file to something relevant.  Would there be a way to parse a more useful file name out of the user input?

**Comment by dhruvmanila** [comment]
FYI, I am not a big fan of the way this code is written just to fool the bot to remove the `require ...` labels. Also, if possible please write tests as the code could break in a matter of months depending on the API used. Take a look at how I wrote it for `fetch_github_info`: https://github.com/TheAlgorithms/Python/blob/master/web_programming/test_fetch_github_info.py

**Comment by xcodz-dot** [comment]
this is more of a command line tool, please implement the following things:
* function is used to get the data (with doctests, and typehints)
* no print or input function is used in function
* file is returned through function and not saved
* You can still have the input output (command line interface) in the `if __name__ == "__main__"` block

File: web_programming/instagram_video.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import json
+from datetime import datetime
+
+import requests
+
+if __name__ == "__main__":
+    link = input("Enter Video/IGTV url: ").strip()
+    url = f"https://downloadgram.net/wp-json/wppress/video-downloader/video?url={link}"
```
Comment (xcodz-dot) [inline web_programming/instagram_video.py:8]:
```suggestion
def download_video(url: str) -> bytes:
    url = f"https://downloadgram.net/wp-json/wppress/video-downloader/video?url={url}"
```

File: web_programming/instagram_video.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    req = requests.get(url)
+    res = req.content
+    json_data = json.loads(res)
+    print(f"Downloading video from {link} ...")
```
Comment (xcodz-dot) [inline web_programming/instagram_video.py:12]:
```suggestion
```

File: web_programming/instagram_video.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    video_data = requests.get(json_data[0]["urls"][0]["src"]).content
+    file_name = f"{datetime.now():%Y-%m-%d_%H:%M:%S}.mp4"
+    with open(file_name, "wb") as fp:
+        fp.write(video_data)
+    print(f"Done. Video saved to disk as {file_name}.")
```
Comment (xcodz-dot) [inline web_programming/instagram_video.py:17]:
```suggestion
    video_data = requests.get(json_data[0]["urls"][0]["src"]).content
    return video_data
```

**Review by xcodz-dot — CHANGES_REQUESTED** [review]
Please make the following changes first then you can add the doctests (or you can leave them because it is a downloading proggram and doctest might not be possible here)

File: web_programming/instagram_video.py (hunk 3 above)
Comment (cclauss) [reply web_programming/instagram_video.py:8]: My sense is that it is too confusing for the readers if we reuse `url` as both a function parameter and a local variable.

**Comment by Epic-R-R** [comment]
I Update the code but in pre-commit, I have some error in the download_video function.
I use isort and black on code and it's ok

**Review by cclauss — APPROVED** [review]
Nice!

**Comment by dhruvmanila** [comment]
So, no tests?

**Comment by Epic-R-R** [comment]
Great, thank you for Update the code

**Comment by xcodz-dot** [comment]
Wow, the code lines shrank too much 😄

**Comment by cclauss** [comment]
;-). That’s what I do... https://github.com/cclauss/Ten-lines-or-less

**Comment by dhruvmanila** [comment]
***Explicit is better than implicit.*** 😉

**Comment by cclauss** [comment]
What is implicit in this code?

**Comment by dhruvmanila** [comment]
Nothing, just putting this line out there :)

**Comment by cclauss** [comment]
[_Namespaces are one honking great idea -- let's do more of those!_](https://www.python.org/dev/peps/pep-0020/)

**Comment by cclauss** [comment]
> So, no tests?

As I wrote you before, web programming tests are difficult to write without mocking so that basically makes doctests a nonstarter.  If someone would be willing to add tests to this or another web programming algorithm, then we could use that as a guide for future web programming contributors.

**Comment by dhruvmanila** [comment]
I will try to do that. In the meantime, if possible can you give a small glance to #3977 just to determine whether it's worth doing?

Also, I am in the middle of redesigning the parser for the bot and it will do more than just a comment. :)

**Pull request merged**